File: shopping-list.cpp
```
#include <iostream>
#include <vector>
#include <cmath>
using namespace std;
int main() {
    vector<float> prices;
    vector<float> discounts;
    float total = 0.0;
    int n, i;
    cin >> n;
    for (i = 0; i < n; i++) {
        float price, discount;
        cin >> price >> discount;
        prices.push_back(price);
        discounts.push_back(discount);
    }
    for (i = 0; i < n; i++) {
        total += prices[i] * (1 - discounts[i] / 100.0);
    }
    cout << fixed << setprecision(2) << total << endl;
    return 0;
}
```

File: shopping-list.py
```python
n = int(input())
prices = list(map(float, input().split()))
discounts = list(map(float, input().split()))

if len(prices) != len(discounts):
    print("Error: The lengths of prices and discounts should be the same.")
else:
    total_price = sum(price * (1 - discount / 100) for price, discount in zip(prices, discounts))
    print(total_price)
```